### PR TITLE
fix(Notification): reserve close button spacing for description-only notices

### LINKED ISSUE #57527: Notification: Close Button overlaps description when title is not provided
### Reproduction link

[https://ant.design/components/notification](https://ant.design/components/notification)

### Steps to reproduce

Using notification.open without providing a title. Go to official antd docs, and delete any title in any notification e.g.

Image

### What is expected?

- The close button should not overlap content. 
- The layout should remain consistent regardless of whether message is provided. 
- Adequate spacing (padding/margin) should always be reserved for the close button.

### What is actually happening?

When message is omitted:

- The close button overlaps the description text
- There is no reserved spacing on the right side
- The issue becomes more noticeable with longer/multiline text

| Environment | Info |
| --- | --- |
| antd | 6.3.5 |
| React | 19.2.4 |
| System | Windows |
| Browser | Chrome Version 146.0.7680.165 |

### PR DESCRIPTION
**Pull request by EndlessLucky**
### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [x] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [x] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ⌨️ Accessibility improvement
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Fixes notification layout overlap when notification.open is called without title/message.
> - Fixes #57527 

### 💡 Background and Solution

When title/message is omitted and only description is rendered, the close button can overlap notification content because right-side spacing is only reserved for the title block.
This PR updates notification styles to reserve close-button spacing for description content in closable notices as well, so layout remains consistent whether title exists or not.
Also added a regression test for the title: null + description case to prevent future regressions.

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Notification close button overlapping description when title is empty by reserving spacing for description-only notices. ✅ Add regression test for description-only closable notification layout.  |
| 🇨🇳 Chinese |  修复 Notification 在 title 为空时关闭按钮与描述内容重叠的问题，为仅描述内容场景预留关闭按钮间距。 ✅ 新增仅描述内容且可关闭场景的回归测试。|

### CODE REVIEW

**Comment by EndlessLucky** [comment]
@afc163 Would you please review this PR? Thanks.

File: components/notification/__tests__/index.test.tsx [hunk 1]
```diff
@@ -511,6 +511,24 @@ describe('notification', () => {
...
+    await awaitPromise();
+
+    expect(document.querySelector('.ant-notification-notice-close')).toBeTruthy();
+    expect(document.querySelector('.ant-notification-notice-description')).toBeTruthy();
+    expect(
+      Array.from(document.querySelectorAll('style'))
+        .map((node) => node.textContent)
+        .join(''),
+    ).toContain('.ant-notification-notice-closable .ant-notification-notice-description');
+  });
```
Comment (QDyanbing) [inline components/notification/__tests__/index.test.tsx]: This kind of test case that only checks the className doesn't seem to have much practical value.We could add a small debug demo and use image diff for regression testing.

File: components/notification/style/index.ts [hunk 2]
```diff
@@ -181,6 +181,10 @@ export const genNoticeStyle: GenerateStyle<NotificationToken, CSSObject> = (toke
       paddingInlineEnd: token.paddingLG,
     },
 
+    [`${noticeCls}-closable ${noticeCls}-description`]: {
+      paddingInlineEnd: token.paddingLG,
+    },
+
```
Comment (QDyanbing) [inline components/notification/style/index.ts]: This fixes the overlapping issue, but it also leaves a fairly large blank space on the right, which makes the layout look less appealing. That's also why this was labeled as a design issue.

File: components/notification/style/index.ts
Comment (afc163) [inline components/notification/style/index.ts:183]:
```suggestion
[`${noticeCls}-description`]: {
  fontSize,
  color: colorText,
  marginTop: token.marginXS,
  
  [`${noticeCls}-description:first-child`]: {
    marginTop: 0,
    marginInlineEnd: token.marginSM,
  },
},
```

File: components/notification/style/index.ts [hunk 3]
```diff
@@ -175,6 +175,11 @@ export const genNoticeStyle: GenerateStyle<NotificationToken, CSSObject> = (toke
       fontSize,
       color: colorText,
       marginTop: token.marginXS,
+      
+      [`${noticeCls}-description:first-child`]: {
```
Comment (afc163) [inline components/notification/style/index.ts]:
```
'&:first-child': {
```

File: components/notification/style/index.ts [hunk 4]
```diff
@@ -175,12 +175,22 @@ export const genNoticeStyle: GenerateStyle<NotificationToken, CSSObject> = (toke
...
+        marginTop: 0,
+        marginInlineEnd: token.marginSM,
+      },
```
Comment (afc163) [inline components/notification/style/index.ts:182]: You can revert all other changes but this one.

**Comment by EndlessLucky** [comment]
@afc163 I fixed the feedback. Would you please review? Thanks.

**Comment by afc163** [comment]
You can revert all other changes but this one.

**Comment by EndlessLucky** [comment]
@afc163 Would you please review? Thanks.

**Review by afc163 — APPROVED** [review]

**Comment by EndlessLucky** [comment]
@afc163 Thank you.
